File: src/Functions.Worker.Extensions.OpenAI/Embeddings/EmbeddingsContext.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

using OpenAI.Embeddings;

namespace Microsoft.Azure.Functions.Worker.Extensions.OpenAI.Embeddings;

public class EmbeddingsContext
{
    public EmbeddingsContext(IList<string> Request, OpenAIEmbeddingCollection? Response)
    {
        this.Request = Request;
        this.Response = Response;
    }

    /// <summary>
    /// Embeddings request sent to OpenAI.
    /// </summary>
    public IList<string> Request { get; set; }

    /// <summary>
    /// Embeddings response from OpenAI.
    /// </summary>
    public OpenAIEmbeddingCollection? Response { get; set; }

    /// <summary>
    /// Gets the number of embeddings that were returned in the response.
    /// </summary>
    public int Count => this.Response?.Count ?? 0;
}

File: java-library/src/main/java/com/microsoft/azure/functions/openai/annotation/embeddings/EmbeddingsContext.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.microsoft.azure.functions.openai.annotation.embeddings;

import com.azure.ai.openai.models.Embeddings;
import java.util.List;

public class EmbeddingsContext {

    private List<String> request;
    private Embeddings response;
    private int count = 0;

    public List<String> getRequest() {
        return request;
    }

    public void setRequest(List<String> request) {
        this.request = request;
    }

    public Embeddings getResponse() {
        return response;
    }

    public void setResponse(Embeddings response) {
        this.response = response;
    }

    /**
     * Gets the number of embeddings that were returned in the response.
     *
     * @return The number of embeddings that were returned in the response.
     */
    public int getCount() {
        return this.response != null && this.response.getData() != null
                ? this.count
                : 0;
    }

}
